orderer/common/multichannel/metrics.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package multichannel
+
+import (
+	"github.com/hyperledger/fabric/common/metrics"
+	"github.com/hyperledger/fabric/orderer/common/types"
+)
+
+var (
+	clusterStatusOpts = metrics.GaugeOpts{
+		Namespace:    "cluster",
+		Subsystem:    "",
+		Name:         "status",
+		Help:         "The status of the cluster node: 0 if inactive, 1 if active, 2 if onboarding.",
+		LabelNames:   []string{"channel"},
+		StatsdFormat: "%{#fqname}.%{channel}",
+	}
+
+	clusterRelationOpts = metrics.GaugeOpts{
+		Namespace:    "cluster",
+		Subsystem:    "",
+		Name:         "relation",
+		Help:         "The cluster relation of the node: 0 if none, 1 if member, 2 if follower, 3 if config-tracker.",
+		LabelNames:   []string{"channel"},
+		StatsdFormat: "%{#fqname}.%{channel}",
+	}
+)
+
+// Metrics defines the metrics for the cluster.
+type Metrics struct {
+	ClusterStatus   metrics.Gauge
+	ClusterRelation metrics.Gauge
+}
+
+// A MetricsProvider is an abstraction for a metrics provider. It is a factory for
+// Counter, Gauge, and Histogram meters.
+type MetricsProvider interface {
+	// NewCounter creates a new instance of a Counter.
+	NewCounter(opts metrics.CounterOpts) metrics.Counter
+	// NewGauge creates a new instance of a Gauge.
+	NewGauge(opts metrics.GaugeOpts) metrics.Gauge
+	// NewHistogram creates a new instance of a Histogram.
+	NewHistogram(opts metrics.HistogramOpts) metrics.Histogram
+}
+
+//go:generate mockery -dir . -name MetricsProvider -case underscore -output ./mocks/
+
+// NewMetrics initializes new metrics for the cluster infrastructure.
+func NewMetrics(m MetricsProvider) *Metrics {
+	return &Metrics{
+		ClusterStatus:   m.NewGauge(clusterStatusOpts),
+		ClusterRelation: m.NewGauge(clusterRelationOpts),
+	}
+}
+
+func (m *Metrics) reportClusterStatus(channel string, status types.Status) {
+	var s int
+	switch status {
+	case types.StatusInactive:
+		s = 0
+	case types.StatusActive:
+		s = 1
+	case types.StatusOnBoarding:
+		s = 2
+	default:
+		logger.Panicf("Programming error: unexpected cluster status %s", status)
+	}
+	m.ClusterStatus.With("channel", channel).Set(float64(s))
+}
+
+func (m *Metrics) reportClusterRelation(channel string, relation types.ClusterRelation) {
+	var r int
+	switch relation {
+	case types.ClusterRelationNone:
+		r = 0
+	case types.ClusterRelationMember:
```

Comment:
wasn't this supposed to be `types.ClusterRelationConsenter`? probably needs a rebase as well

Reply:
Done